### DISCUSSION TITLE
Is there a way to IP whitelist access to Vercel URLs?

### DISCUSSION THREAD

**Discussion by jaeyow** [vercel/vercel#6501] [Help] [answered]
After a build we get an email from a vercel bot like the following:

Successfully deployed to the following URLs:

my-next-app-gitrepo.vercel.app
my-next-app.vercel.app
my-next-app-git-main-gitrepo.vercel.app

If this email somehow gets out, people who have access to these URLs will be able to access the site without any issues. However this is a problem because this is intended to be just internal URLs and not meant to be for public. Say the site has confidential content and not ready for prime time yet?

Have you encountered this issue and how did you get around it? 

Thanks,

**Comment by jaeyow** [comment, marked as answer]
After looking a little bit more on this, found that Vercel does make this possible with their [Password-protect Vercel Deployments](https://vercel.com/blog/protecting-deployments), however at $150 per month + $20 per month on the Pro Plan - that's a bit steep, makes one want to think twice. Happy to hear any other options out there.

**Reply by leerob** [reply]
You can use this. It's not the same as IP allowlisting, but it does protect deployments!

https://github.com/storyofams/next-password-protect

**Reply by jaeyow** [reply]
thanks @leerob I will definitely give that a try. Yes not exactly the same as IP whitelisting, but will prevent my website from being viewed when not yet ready, as well as probably keep web crawlers at bay, since my content although eventually will be made public, is hush hush right now until after its go live date.
